### DISCUSSION TITLE
install gatsby for first time receiving npm warn

### DISCUSSION THREAD

**Discussion by Brandon-B91** [gatsbyjs/gatsby#30711] [Help]
installing gatsby for first time getting some weird npm warns

node version is 15.12.0
npm version 7.7.5
gatsby -v shows Gatsby CLI version: 3.1.0

this is the error 

brandonbrown@Brandons-MacBook-Air ~ % npm install -g gatsby-cli
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: graphql@15.5.0
npm WARN node_modules/gatsby-cli/node_modules/graphql
npm WARN   graphql@"^15.4.0" from gatsby-recipes@0.12.0
npm WARN   node_modules/gatsby-cli/node_modules/gatsby-recipes
npm WARN     gatsby-recipes@"^0.12.0" from gatsby-cli@3.1.0
npm WARN     node_modules/gatsby-cli
npm WARN   2 more (@graphql-tools/schema, @graphql-tools/utils)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer graphql@"^14.4.1" from express-graphql@0.9.0
npm WARN node_modules/gatsby-cli/node_modules/express-graphql
npm WARN   express-graphql@"^0.9.0" from gatsby-recipes@0.12.0
npm WARN   node_modules/gatsby-cli/node_modules/gatsby-recipes
npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'

added 575 packages, and audited 576 packages in 18s

105 packages are looking for funding
  run `npm fund` for details

i did open vscode and gatsby new seems to work just fine so I dont know

**Comment by backbencher00** [comment]
i wanna work on this

**Comment by jomarmontuya** [comment]
I'm also getting the same error Is this a global outage with NPM?

**Comment by Brandon-B91** [comment]
Intersting well glad to know I'm now the only one lol.

**Comment by jsn-t** [comment]
I can't boot my project at all due to this issue, hopefully this gets fixed

**Comment by jomarmontuya** [comment]
I downgraded my nodejs to version 14 and it works pretty well

**Comment by Brandon-B91** [comment]
So it looks to be an issue with node version 15 then.

**Comment by LekoArts** [comment]
Node 15 uses npm 7 which has messed up peerDependency tracking (and thus introduced the `--legacy-peer-deps` flag for installs). So when you downgrade to Node 14 you get npm 6 and the old behavior. For now you'll need to use that flag when using npm 7, see: https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#updating-community-plugins
